src/controllers/spreadsheetCommand.ts

```diff
@@ -0,0 +1,256 @@
+import { Logger } from 'structured-log';
+import * as _ from 'lodash';
+import * as moment from 'moment';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import * as discord from 'discord.js';
+import { IActiveCollabAPI } from './activecollab-api';
+import { ICommandController } from './command';
+import { TimeRecord } from '../models/timeRecords';
+import { promisify } from 'util';
+
+const spreadsheetPath: string = 'Spreadsheets';
+
+export async function filteredTasks(
+    nameFilters: string[],
+    projectFilters: string[],
+    startDate: string,
+    endDate: string,
+    eventColor: any,
+    activeCollabApi: IActiveCollabAPI,
+    logger: Logger,
+    writeToCsv: (
+        columnNames: string[],
+        rows: string[][],
+        filePath: string,
+        logger: Logger
+    ) => Promise<void>
+): Promise<discord.RichEmbed> {
+    const message = new discord.RichEmbed();
+
+    let tasks: _.LoDashImplicitArrayWrapper<TimeRecord>;
```

Comment:
It seems possible to me to do all required functionality here without lodash, I may be wrong though.